src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -42,13 +42,86 @@ namespace ErrorCodes
     extern const int DATABASE_ACCESS_DENIED;
     extern const int NOT_IMPLEMENTED;
     extern const int LOGICAL_ERROR;
+    extern const int BAD_ARGUMENTS;
 }
 
 namespace StorageObjectStorageSetting
 {
 extern const StorageObjectStorageSettingsBool allow_dynamic_metadata_for_data_lakes;
 }
 
+namespace
+{
+    void sanityCheckPartitioningConfiguration(
```

Comment:
IMO this whole function (and getPartitionByAst) can be moved inside `PartitionStrategyProvider`, since those only validate/prepare parameters for `PartitionStrategyProvider::get`.